MathLib/InterpolationAlgorithms/PiecewiseLinearInterpolation.cpp

```diff
@@ -52,13 +52,18 @@ double PiecewiseLinearInterpolation::getValue(double pnt_to_interpolate) const
                                     pnt_to_interpolate));
     std::size_t const interval_idx = std::distance(_supp_pnts.begin(), it) - 1;
 
+    // support points.
+    double const x = _supp_pnts[interval_idx];
+    double const x_r = _supp_pnts[interval_idx + 1];
+
+    // values.
+    double const f = _values_at_supp_pnts[interval_idx];
+    double const f_r = _values_at_supp_pnts[interval_idx + 1];
+
     // compute linear interpolation polynom: y = m * (x - support[i]) + value[i]
-    const double m((_values_at_supp_pnts[interval_idx + 1] -
-                    _values_at_supp_pnts[interval_idx]) /
-                   (_supp_pnts[interval_idx + 1] - _supp_pnts[interval_idx]));
+    const double m = (f_r - f) / (x_r - x);
 
-    return m * (pnt_to_interpolate - _supp_pnts[interval_idx]) +
-           _values_at_supp_pnts[interval_idx];
+    return m * (pnt_to_interpolate - x) + f;
 }
 
 double PiecewiseLinearInterpolation::getDerivative(
```

Comment:
It would be good if the implementation would be (re)moved from the class since it neither computes the derivative nor it computes the gradient of a piecewise linear function.